### DISCUSSION TITLE
Error while importing wbi_core

### DISCUSSION THREAD

**Discussion by versant2612** [LeMyst/WikibaseIntegrator#353] [General]
Successfully installed ujson-5.2.0 wikibaseintegrator-0.12.0rc4
....

from wikibaseintegrator import wbi_core

ImportError: cannot import name 'wbi_core' from 'wikibaseintegrator' (/usr/local/lib/python3.9/dist-packages/wikibaseintegrator/__init__.py)

**Comment by LeMyst** [comment]
WikibaseIntegrator v0.12 is a huge rewrite of the v0.11.
You can find more documentation in the dedicated branch README:
https://github.com/LeMyst/WikibaseIntegrator/blob/rewrite-wbi/README.md

You also have the documentation https://wikibaseintegrator.readthedocs.io/

And you have example in the notebooks of the branch:
https://github.com/LeMyst/WikibaseIntegrator/tree/rewrite-wbi/notebooks

**Comment by versant2612** [comment]
Thanks again, the notebook examples helped a lot.
I'm changing my python script from WikidataIntegrator to WikibaseIntegrator.

Do you have examples of claims with qualifiers and references?

**Reply by LeMyst** [reply]
I've updated the notebook:
https://github.com/LeMyst/WikibaseIntegrator/blob/rewrite-wbi/notebooks/item_create_new.ipynb

**Reply by versant2612** [reply]
Help me a lot again !!!

**Comment by versant2612** [comment]
I also need some help to debug errors such as 

ERROR:root:Error while writing to the Wikibase instance

**Reply by LeMyst** [reply]
Do you have a stacktrace and/or the code used?

**Reply by versant2612** [reply]
For example, for this case, I used the strip() string function for **every** string. Because I don't know which string field from the imput file raised this error. 

ERROR:root:Error while writing to the Wikibase instance
Erro ao gravar:  A learning project : the development of sustainable support in the use of instructional technology.
("{'error': {'code': 'modification-failed', 'info': 'Malformed input: "
 "McAlpine, Lynn (Supervisor)', 'messages': [{'name': "
 "'wikibase-validator-malformed-value', 'parameters': [' McAlpine, Lynn "
 "(Supervisor)'], 'html': {'*': 'Entrada malformada:  McAlpine, Lynn "
 "(Supervisor)'}}], '*': 'See http://localhost/w/api.php for API usage. "
 'Subscribe to the mediawiki-api-announce mailing list at '
 '&lt;https://lists.wikimedia.org/mailman/listinfo/mediawiki-api-announce&gt; '
 "for notice of API deprecations and breaking changes.'}}")
Traceback (most recent call last):
  File "/home/veronicasantos/loadNDTLD.py", line 180, in createEntity
    e1_item.write()
  File "/usr/local/lib/python3.9/dist-packages/wikibaseintegrator/entities/item.py", line 107, in write
    json_data = super()._write(data=self.get_json(), **kwargs)
  File "/usr/local/lib/python3.9/dist-packages/wikibaseintegrator/entities/baseentity.py", line 241, in _write
    json_result: dict = mediawiki_api_call_helper(data=payload, login=login, allow_anonymous=allow_anonymous, is_bot=is_bot, **kwargs)
  File "/usr/local/lib/python3.9/dist-packages/wikibaseintegrator/wbi_helpers.py", line 206, in mediawiki_api_call_helper
    return mediawiki_api_call('POST', mediawiki_api_url=mediawiki_api_url, session=session, data=data, headers=headers, max_retries=max_retries, retry_after=retry_after, **kwargs)
  File "/usr/local/lib/python3.9/dist-packages/wikibaseintegrator/wbi_helpers.py", line 120, in mediawiki_api_call
    raise MWApiError(response.json() if response else {})
wikibaseintegrator.wbi_exceptions.MWApiError: {'error': {'code': 'modification-failed', 'info': 'Malformed input: McAlpine, Lynn (Supervisor)', 'messages': [{'name': 'wikibase-validator-malformed-value', 'parameters': [' McAlpine, Lynn (Supervisor)'], 'html': {'*': 'Entrada malformada:  McAlpine, Lynn (Supervisor)'}}], '*': 'See http://localhost/w/api.php for API usage. Subscribe to the mediawiki-api-announce mailing list at &lt;https://lists.wikimedia.org/mailman/listinfo/mediawiki-api-announce&gt; for notice of API deprecations and breaking changes.'}}

**Reply by versant2612** [reply]
Very often this error message appears and no exception is generated

ERROR:root:Error while writing to the Wikibase instance
ERROR:root:Error while writing to the Wikibase instance
ERROR:root:Error while writing to the Wikibase instance
ERROR:root:Error while writing to the Wikibase instance
ERROR:root:Error while writing to the Wikibase instance

```
....
        if OK_Publication:                

            e1_item = wbi.item.new()
            e1_item.claims.add(e1_statements)
            e1_item.labels.set('en', title_text[0:249].strip())
            if title_text[0:249].strip() != long_description[0:249].strip(): 
                e1_item.descriptions.set('en', long_description[0:249].strip())

            try:
                e1_item.write()
                total_gravados = total_gravados + 1
            except Exception as e:
                if 'already has label' in str(e): 
                      ...
                else:
                    print('Erro ao gravar: ', str(e))

...
```

**Reply by dpriskorn** [reply]
> Very often this error message appears and no exception is generated
> 
> ERROR:root:Error while writing to the Wikibase instance ERROR:root:Error while writing to the Wikibase instance ERROR:root:Error while writing to the Wikibase instance ERROR:root:Error while writing to the Wikibase instance ERROR:root:Error while writing to the Wikibase instance

I also get this, and I think it is a bug. Would you be willing to open a new issue and state which version you are using?

**Reply by LeMyst** [reply]
Hello @versant2612 ,
Sorry for the late answer.
For your first issue, I think you try to put string data into a non-"String" property, like an Item property or Quantity, can you check if you use the correct property, and if it's the case, which type ?
For your second message, I've improved exception logging in the commit fd391aa2998b7831b406e5cf19a7f29b42e851a5 of branch rewrite-wbi.

**Reply by versant2612** [reply]
I found the error source: a property of type URL but in the data source it was as text. But I believe it would help if the message indicated which property raised the error.

I'll update and test the logging update.

**Reply by LeMyst** [reply]
It would be nice, but it's a problem on the Wikibase side.

**Reply by versant2612** [reply]
> > Very often this error message appears and no exception is generated
> > ERROR:root:Error while writing to the Wikibase instance ERROR:root:Error while writing to the Wikibase instance ERROR:root:Error while writing to the Wikibase instance ERROR:root:Error while writing to the Wikibase instance ERROR:root:Error while writing to the Wikibase instance
> 
> I also get this, and I think it is a bug. Would you be willing to open a new issue and state which version you are using?

Do you think it is a Wikibase bug? 
I used a docker installation, basic components version are bellow

MediaWiki 	1.36.3
PHP 	7.4.27 (apache2handler)
MariaDB 	10.3.34-MariaDB-1:10.3.34+maria~focal
ICU 	67.1
Elasticsearch 	6.8.23
Lua 	5.1.5

**Reply by dpriskorn** [reply]
This bug has not been reported yet to the Wikibase team. Would you like to report it in phabricator?
